commons-numbers-combinatorics/src/main/java/org/apache/commons/numbers/combinatorics/BinomialCoefficient.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.commons.numbers.combinatorics;
+
+import org.apache.commons.numbers.core.ArithmeticUtils;
+
+/**
+ * Representation of the <a href="http://mathworld.wolfram.com/BinomialCoefficient.html">
+ * binomial coefficient</a>.
+ * It is "{@code n choose k}", the number of {@code k}-element subsets that
+ * can be selected from an {@code n}-element set.
+ */
+public final class BinomialCoefficient {
+
+    /** Private constructor. */
+    private BinomialCoefficient() {
+        // intentionally empty.
+    }
+
+    /**
+     * Computes the binomial coefficient.
+     * The largest value of {@code n} for which all coefficients can
+     * fit into a {@code long} is 66.
+     *
+     * @param n Size of the set.
+     * @param k Size of the subsets to be counted.
+     * @return {@code n choose k}.
+     * @throws IllegalArgumentException if {@code n < 0}.
+     * @throws IllegalArgumentException if {@code k > n}.
+     * @throws ArithmeticException if the result is too large to be
+     * represented by a {@code long}.
+     */
+    public static long value(int n, int k) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.